glib/src/array.rs

```diff
@@ -0,0 +1,34 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use crate::translate::*;
+use std::fmt;
+
+wrapper! {
+    #[derive(PartialEq, Eq, PartialOrd, Ord, Hash)]
+    pub struct Array(Shared<ffi::GArray>);
+
+    match fn {
+        ref => |ptr| ffi::g_array_ref(ptr),
+        unref => |ptr| ffi::g_array_unref(ptr),
+        get_type => || ffi::g_array_get_type(),
+    }
+}
+
+impl Array {
+    pub fn len(&self) -> usize {
+        unsafe { (*self.to_glib_none().0).len as usize }
+    }
+
+    pub fn data(&self) -> *mut libc::c_char {
```

Comment:
```suggestion
    pub fn data(&self) -> *mut libc::c_void {
```

Seems more correct

Reply:
Should be unsafe as well

Reply:
No, just like `as_ptr()` or `into_raw()` is not unsafe. All usage that would cause problems requires unsafe code, getting the pointer itself is not a problem.

Reply:
*mut or *const is preferrable?

Reply:
`mut` seems fine